improver_tests/calibration/dataframe_utilities/test_dataframe_utilities.py

```diff
@@ -762,6 +801,83 @@ def test_old_site_with_only_one_forecast_and_truth(self):
         self.assertCubeEqual(result[0], self.expected_period_forecast)
         self.assertCubeEqual(result[1], self.expected_period_truth)
 
+    def test_station_id_new_site_with_only_one_forecast(self):
+        """Test for a site that has a forecast and truth data point for the most
+        recent time only. Other sites are present at all forecast and truth times.
+        Only the forecast DataFrame has a station_id column. This mimics the
+        situation when a new site is added to the forecast and truth
+        DataFrames. The forecast and truth cubes generated have three sites
+        with the 'new' site having NaNs for all time points except for the most
+        recent time. The resulting forecast cube has a station_id coordinate."""
+        self.forecast_df_multi_station_id["station_id"] = (
+            self.forecast_df_multi_station_id["wmo_id"] + "0"
+        )
```

Comment:
It looks like this `station_id` column is added in all the tests that use `self.forecast_df_multi_station_id`, and it is always defined the same way, so maybe this could be done in the setup instead. (Similarly for `self.truth_df_multi_station_id`.)

Reply:
I've centralised this as suggested.